Frameworks/UIKit/StarboardXaml/ApplicationCompositor.cpp

```diff
@@ -53,12 +53,31 @@ void InitializeApp() {
     SetupMainRunLoopTimedMultipleWaiter();
 }
 
+extern "C" void RunApplicationMainWithString(Platform::String^ principalClassName,
+                                   Platform::String^ delegateClassName,
+                                   float windowWidth,
+                                   float windowHeight,
+                                   ActivationType activationType,
+                                   Platform::String^ activationString) {
+    // Perform initialization
+    InitializeApp();
+
+    // Kick off iOS application main startup
+    ApplicationMainStartHSTRING(
```

Comment:
Why do we need this? Can't we just use a void* and cast it based on the activationType?